File: searchnsort/selection_sort.cpp
//AlgoExpert - selection sort - sorting

#include<vector>
using namespace std;

vectot<int> selectionSort(vector<int> array);

vector<int> selectionSort(vector<int> array){
	if (array.empty()){
		return {};
	}
	int startIdx = 0;
	while (startIdx < array.size() - 1){
		int smallestIdx = startIdx;
		for (int i = startIdx + 1; i < array.size(); i++){
			if (array[smallestIdx] > array[i]){
				smallestIdx = i;
			}
		}
		swap(array[startIdx], array[smallestIdx]);
		startIdx++;
	}
	return array;
}

File: searchnsort/selection_sort.py
'''
AlgoExpert - selection sort - sorting
'''

def selectionSort(array):
	current_idx = 0
	while current_idx < len(array) - 1:
		smallest_idx = current_idx
		for i in range(current_idx + 1, len(array)):
			if array[smallest_idx] > array[i]:
				smallest_idx = i
		swap(current_idx, smallest_idx, array)
		current_idx += 1
	return array

def swap(i, j, array):
	array[i], array[j] = array[j], array[i]
